src/XGrid.cu

```diff
@@ -0,0 +1,260 @@
+/* 
+
+Implements the grid-multiplication onto a GPU using CUDA. 
+
+*/
```

Comment:
Missing the boiler plate stuff at the start of other .cu files. Do we need that?

Also, please give a more descriptive explanation of what this module does.